### PR TITLE
Add all ledgers endpoint

### LINKED ISSUE #57: [endpoint] All ledgers
This endpoint represents all ledgers.

https://www.stellar.org/developers/horizon/reference/endpoints/ledgers-all.html

### PR DESCRIPTION
**Pull request by robertDurst**
Modeled heavily off of All Assets, this endpoint represents the all ledgers endpoint for the stellar horizon server.

Since this is my first PR please critique mercilessly :)

Closes #57

### CODE REVIEW

File: client/src/endpoint/ledger.rs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+/// #
+/// let client      = Client::horizon_test().unwrap();
+/// let endpoint    = AllLedgers::default().limit(3);
+/// let records     = client.request(endpoint).unwrap();
+/// #
+/// # assert_eq!(records.records().len(), 3);
+/// ```
+#[derive(Debug, Default)]
+pub struct AllLedgers {
+    // // TODO: Cursor needs to be parseable from the links that come back
```
Comment (kbacha) [inline client/src/endpoint/ledger.rs]: I may have left this comment in the AllAssets but I think it can be removed now 😄

**Comment by robertDurst** [comment]
Some tests failed, I will fix these up.

**Comment by correlator** [comment]
we also use cargo fmt to lint the repo.  Travis will also fail if there is a diff between your commit and the cargo fmt output. https://github.com/rust-lang-nursery/rustfmt <-- just a heads up.  Again thanks for joining and taking the initiative to dive right in.  This is awesome!

**Comment by kbacha** [comment]
Thanks for jumping in with two feet! This is a great start :)

**Comment by robertDurst** [comment]
Hey so this seems to be the problem I am having: 

`thread 'main' panicked at 'called Result::unwrap() on an Err value: ParseError(ErrorImpl { code: Message("missing field `prev_hash`"), line: 51, column: 7 })'`

And it is related to the doc tests.

**Comment by robertDurst** [comment]
So it appears prev_hash is not actually a value on a ledger, so that was causing tests to fail. Everything passes now!

https://horizon-testnet.stellar.org/ledgers?limit=1

File: resources/src/ledger.rs [hunk 2]
```diff
@@ -37,11 +36,6 @@ impl Ledger {
         &self.hash
     }
 
-    /// The hash of the ledger that chronologically came before this one.
-    pub fn prev_hash(&self) -> &String {
```
Comment (kbacha) [inline resources/src/ledger.rs]: :+1:

**Comment by kbacha** [comment]
@robertDurst I updated some of the ways I documented and arranged the modules in #69 if you want to take a look 😄 

Hopefully those changes make sense and we can match these PRs. I don't mind refactoring yours after we merge though.

**Comment by robertDurst** [comment]
Sounds good! I'll check out this stuff and squash tomorrow!

**Comment by robertDurst** [comment]
Ok.... let me squash and this should be good to go.

**Comment by robertDurst** [comment]
Hey @kbacha so I am not very good at rebasing and squashing 😅 ... I did it once and I'm still not 100% sure I didn't mess it all up.

Essentially here, I assume I would need to move all my commits into one, after the two PR's that I pulled (and were submitted since this PR has been open).

Otherwise I reformatted the ledger docs and I think it all looks good.

**Comment by correlator** [comment]
Try following these steps @robertDurst :

1. Checkout and pull master from GitHub
2. checkout robertDurst:all-ledgers
3. `git rebase master`.  From the looks of it you will have conflicts to resolve.  You can also do `git merge master` here if you are more comfortable resolving conflicts in a merge commit.  Conflict resolution in a rebase is slightly different, so up to you on how many new things you want to try at once.

Now you should be on robertDurst:all-ledgers and be up to date with the changes in master.

This is the most explicit way I know of to squash old commits, there are fancier shortcuts out there you can learn about.
4. type `git status`, git log` and you should see your full commit history
5. `git reset HEAD^ --soft`, `git status`, `git log` and you should see that we just undid the previous commit with all changes still staged.
6. Repeat step 5 until you have all the changes you want stagged for commit.  `git commit` should open a vim editor.  The first line should be the title of your commit followed by a few sentences describing with you did.
7. All your changes are saved with a single commit :beers:

**Comment by kbacha** [comment]
You may want to use interactive rebase.

https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history#squash-commits-together

**Comment by robertDurst** [comment]
Thanks @correlator and @kbacha! So I have squashed all down to one commit.

However it also squashed the merged code I had in the middle of my PR's...

**Comment by kbacha** [comment]
I'm guessing that your merge commit may have pulled in some in-between state. You can probably revert those files by checking them out individually.

```bash
$ git checkout CONTRIBUTING.md
```

And then amending your commit:

```bash
$ git commit --amend
$ git push -f origin HEAD
```

**Comment by kbacha** [comment]
btw, I'll merge this as soon as those extraneous changes are removed 👍

**Comment by robertDurst** [comment]
Learn something new about git everyday... thanks for your patience 👍

**Comment by robertDurst** [comment]
Now it is messed up again. Ill give this another go after work.

**Comment by kbacha** [comment]
i may have made the checkout harder to understand, but if you do:
```bash
git checkout master -- <filename/directory>
```

then I think it works. Basically, this tells git not to checkout the branch but to checkout a particular file instead.

**Comment by robertDurst** [comment]
Got it down to two commits.

**Comment by robertDurst** [comment]
Started celebrating when rebase worked... and then I failed the tests. However, I am unable to reproduce this failure @kbacha. Help would be awesome :)

**Comment by robertDurst** [comment]
Rebase worked 👍  but then build broke 👎 . Having issues reproducing the error. Any help here would be awesome!

**Comment by kbacha** [comment]
@robertDurst I'll rebuild. I wonder if the horizon test api blocked us? It looks like requests timed out.

**Comment by kbacha** [comment]
Thanks @robertDurst !

**Pull request merged**